docusaurus/docs/develop/developer_guide/adding_params.md

```diff
@@ -3,275 +3,302 @@ sidebar_position: 5
+  }
+
+  for paramName, paramValue := range params {
+    switch paramName {
+    case prooftypes.ParamNewParameterName:
+      msgUpdateParams.Params.NewParameterName = uint64(paramValue.value.(int64))
+    default:
+      s.Fatalf("unexpected %q type param name %q", paramValue.typeStr, paramName)
+    }
+  }
+
+  return msgUpdateParams
+}
+```
+
+### 3. Update the Default Parameter Values
+
+In the corresponding Go file (e.g., `params.go`), define the default value for the
+new parameter and include it in the `NewParams` and `DefaultParams` functions.
+
+```go
+var (
+  // Other existing parameters...
+
+  KeyNewParameterName = []byte("NewParameterName")
+  ParamNewParameterName = "new_parameter_name"
+  DefaultNewParameterName uint64 = 100 // Example default value
+)
+
+func NewParams(
+  // Other existing parameters...
+  newParameterName uint64,
+) Params {
+  return Params{
+    // Other existing parameters...
+    NewParameterName: newParameterName,
+  }
+}
+
+func DefaultParams() Params {
+  return NewParams(
+    // Other existing default parameters...
+    DefaultNewParameterName,
+  )
+}
+```
+
+### 4. Add Parameter Default to Genesis Configuration
+
+Add the new parameter to the genesis configuration file (e.g., `config.yml`).
+
+```yaml
+genesis:
+  proof:
+    params:
+      # Other existing parameters...
+
+      new_parameter_name: 100
+```
+
+### 5. Modify the Makefile
+
+Add a new target in the `Makefile` to update the new parameter.
+
+```makefile
+.PHONY: params_update_proof_new_parameter_name
+params_update_proof_new_parameter_name: ## Update the proof module new_parameter_name param
+  poktrolld tx authz exec ./tools/scripts/params/proof_new_parameter_name.json $(PARAM_FLAGS)
+```
+
+### 6. Create a new JSON File for the Individual Parameter Update
+
+Create a new JSON file (e.g., `proof_new_parameter_name.json`) in the appropriate
+directory to specify how to update the new parameter.
+
+```json
+{
+  "body": {
+    "messages": [
+      {
+        "@type": "/poktroll.proof.MsgUpdateParam",
+        "authority": "pokt10d07y265gmmuvt4z0w9aw880jnsr700j8yv32t",
+        "name": "new_parameter_name",
+        "as_int64": "100"
+      }
+    ]
+  }
+}
+```
+
+### 7. Update the JSON File for Updating All Parameters for the Module
+
+Add a line to the existing module's `MsgUpdateParam` JSON file (e.g., `proof_all.json`)
+with the default value for the new parameter.
+
+```json
+{
+  "body": {
+    "messages": [
+      {
+        "@type": "/poktroll.proof.MsgUpdateParams",
+        "authority": "pokt10d07y265gmmuvt4z0w9aw880jnsr700j8yv32t",
+        "params": {
+          "min_relay_difficulty_bits": "0",
+          "proof_request_probability": "0.25",
+          "proof_requirement_threshold": "20",
+          "new_parameter_name": "100" // Add this line
+        }
+      }
+    ]
+  }
+}
+```
+
+### 8. Add Parameter Validation
+
+Implement a validation function for the new parameter in the same Go file.
```

Comment:
#PUC: "the same Go file" as which, params.go? I just read about the makefile and JSON files inbetween reading about params.go and here.